celer/tests/isolate_conv_warn.py

```diff
@@ -0,0 +1,45 @@
+from sklearn.exceptions import ConvergenceWarning
+import numpy as np
+from numpy.linalg import norm
+from sklearn.utils.estimator_checks import check_estimator
+
+from celer.dropin_sklearn import LogisticRegression
+from celer.utils.testing import build_dataset
+
+import warnings
+warnings.filterwarnings("error")
+
+
+def isolate_logreg_warning(sparse_X):
```

Comment:
this design could be simpler. Can you not just put a breakpoint at the convergence warning in homotopy.py, then run the test and see when it breaks, then dump X, y and alpha manually ?